lib/cisco_node_utils/span_session.rb

```diff
@@ -0,0 +1,125 @@
+# Insert appropriate license here
+require_relative 'node_util'
+require_relative 'interface'
+
+
+module Cisco
+  # node utils class for SPAN (switchport analyzer) sessions
+  class SpanSession < NodeUtil
+    attr_reader :session_id
+
+    def initialize(session_id)
+      validate_args(session_id)
+    end
+
+    def self.sessions
+      hash = {}
+      all = config_get('span_session', 'all_sessions')
+      return hash if all.nil?
+
+      all.each do |id|
+        id = id.downcase
+        hash[id] = SpanSession.new(id)
+      end
+      hash
+    end
+
+    def validate_args(session_id)
+      fail TypeError unless session_id.is_a?(Integer)
+      fail ArgumentError unless session_id >= 1 || session_id <= 32
+      @session_id = session_id.downcase
+      set_args_keys
```

Comment:
I don't see where you are defining `set_args_keys`.  If you want to use it here are some examples from other objects where we define it for use within the object.

https://github.com/cisco/cisco-network-node-utils/blob/45aaac5fb9098b7a9d8322268e390b25ed324c05/lib/cisco_node_utils/vxlan_vtep_vni.rb#L65

The main purpose of this helper method is to create a list of base keys that are used for all properties or most properties in an object and then add property specific keys to the list of base keys as needed and then reset back to the base list.

This object IMO probably does not need it but I will leave that up to you.